gtsam_unstable/slam/SmartProjectionPoseFactorRollingShutter.h

```diff
@@ -0,0 +1,438 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   SmartProjectionPoseFactorRollingShutter.h
```

Comment:
Can we add a .cpp file to make this file easier to read?

Reply:
this is templated. What do you suggest moving?

Reply:
Ah, never mind.

Reply:
happy to template the basic (non-smart factor) if you like, to keep symmetry.